src/core/derivation.ts

```diff
@@ -181,9 +186,25 @@ export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T, con
     }
     globalState.trackingDerivation = prevTracking
     bindDependencies(derivation)
+
+    if (derivation.observing.length === 0) {
+        warnAboutDerivationWithoutDependencies(derivation)
+    }
+
     return result
 }
 
+function warnAboutDerivationWithoutDependencies(derivation: IDerivation) {
+    if (process.env.NODE_ENV === "production") return
+    if (globalState.reactionRequiresObservable || derivation.requiresObservable) {
```

Comment:
fail on `derivation.requiresObservable` like in computed? https://github.com/mobxjs/mobx/blob/2bce97d1e60ae819c25c8f8b04c764f40855e4f6/src/core/computedvalue.ts#L272-L273

Reply:
`computed` is different, it throws the error to the reading context.

 autorun/reaction wrapped in the mobx error boundaries,
I've actually played with `fail` and i'm really not sure what should be the behaviour.
So for simplicity i think it better be that way